src/structure/options.ts

```diff
@@ -199,6 +199,11 @@ export class StructureOptions extends OptionsGroup {
         // make the settings modal draggable
         makeDraggable(modalDialog, '.modal-header');
 
+        // Stop progation when pressing a key (Jupyter widget)
+        modal.addEventListener('keydown', (event) => {
+            event.stopPropagation();
+        });
```

Comment:
```suggestion
        modal.addEventListener('keydown', (event) => event.stopPropagation());
```

and same for the other modals!